extensions/cms/src/cmsResource/cmsResourceCommands.ts

```diff
@@ -76,16 +77,17 @@ export function addRegisteredServerCommand(appContext: AppContext, tree: CmsReso
 		let relativePath = node instanceof CmsResourceTreeNode ? '' : node.relativePath;
 		let serverName = node instanceof CmsResourceTreeNode ? node.connection.options.registeredServerName === ''
 			? node.connection.options.server : node.connection.options.registeredServerName : null;
-		await appContext.cmsUtils.addRegisteredServer(relativePath, node.ownerUri, serverName).then((result) => {
+		try {
+			let result = await appContext.cmsUtils.addRegisteredServer(relativePath, node.ownerUri, serverName);
 			if (result) {
 				tree.notifyNodeChanged(node);
 			}
-		}, (error) => {
+		} catch (error) {
 			// error out
-			let errorText = localize('cms.errors.addRegisterServerFail', 'Could not add the Registered Server {0}', error);
+			let errorText = error ? error.message : localize('cms.errors.addRegisterServerFail', 'Could not add the Registered Server {0}', error);
 			appContext.apiWrapper.showErrorMessage(errorText);
-			return;
-		});
+			return Promise.reject(new Error(errorText));
+		}
```

Comment:
Just rethrow the error so you don't lose the original context of where the error occurred